region/maridia/outer/Crab Hole.json

```diff
@@ -385,6 +232,57 @@
         }
       }
     },
+    {
+      "link": [1, 4],
+      "name": "Carry G-Mode Through Tube (Left to Right)",
+      "notable": false,
+      "entranceCondition": {
+        "comeInWithGMode": {
+          "mode": "any",
+          "morphed": false
+        }
+      },
+      "requires": [],
+      "exitCondition": {
+        "leaveWithGMode": {
+          "morphed": false
+        }
+      }
+    },
+    {
+      "link": [1, 4],
+      "name": "Carry G-Mode Morph Through Tube (Left to Right)",
+      "notable": false,
+      "entranceCondition": {
+        "comeInWithGMode": {
+          "mode": "any",
+          "morphed": true
+        }
+      },
+      "requires": [
+        {"or": [
+          "Morph",
+          "SpringBall",
+          {"and": [
+            "Gravity",
+            "h_canArtificialMorphIBJ"
+          ]},
+          {"and": [
+            "Gravity",
+            "h_canArtificialMorphBombHorizontally",
```

Comment:
Similar to the strat going the other direction, It's possible to save a PB by rolling on the platform with Gravity turned off. This direction is trickier because you need to wait a moment for the crab to move, and since you're stopping I found you have to move back toward the transition a bit in order to have space to gain enough momentum to make it.